src/display/annotation_layer.js

```diff
@@ -570,6 +570,7 @@ class LinkAnnotationElement extends AnnotationElement {
   render() {
     const { data, linkService } = this;
     const link = document.createElement("a");
+    link.setAttribute("id", data.id);
```

Comment:
Why is this necessary, since all containers already provide that information in a `data-annotation-id` attribute and I cannot immediately tell where in this patch (or the tests) this is even being used?

Reply:
We "need" it here:
https://github.com/mozilla/pdf.js/blob/b9f5a70dede81bd75a6381b0fcba50b5f4e7cf9d/web/pdf_scripting_manager.js#L354

Most of (all?) the actions have an impact on the field's HTML counterpart (e.g. textarea/input for a text field) so it's why I originally decided to use the `id`: my idea was to directly have the element which will receive the change.
And I think it wasn't so bad.
I added the `id` on the `a` just to be consistent with what we've for other containers.

Reply:
Actually, the potentially annoying part of adding `id`s to these elements is that they become "linkable" through the URL hash in a way that's not supported/intended (see issue #11499 and PR #11503 for some context).

These `id`s *could* potentially clash with named destinations, and by adding this to all link elements these issues could very well become more common. I wonder if there's a better way to handle things rather than using `id`s!?

Reply:
At least from a JS pov, I think we can get rid of it easily: we could select the `section` with the correct `data-annotation-id` which will itself update the correct child.
I wonder if it could have a measurable impact on perf: query selector vs getElementById and then go to the child...
Anyway I played `/f1040.pdf#208R` (an annotation on the bottom of page 2) and the form is scrolled...

Reply:
> I wonder if it could have a measurable impact on perf: query selector vs getElementById and then go to the child...

My gut feeling is that query-selector probably is slower in general :-(

Although we're already using that some parts of the AnnotationLayer-code, see https://github.com/mozilla/pdf.js/blob/b9f5a70dede81bd75a6381b0fcba50b5f4e7cf9d/src/display/annotation_layer.js#L2506-L2508 and https://github.com/mozilla/pdf.js/blob/b9f5a70dede81bd75a6381b0fcba50b5f4e7cf9d/src/display/annotation_layer.js#L2553 

> Anyway I played `/f1040.pdf#208R` (an annotation on the bottom of page 2) and the form is scrolled...

Which it obviously shouldn't :-)
As mentioned that may cause some trouble if the document has a "named destination" which happens to be called 208R.

---

Anyway, we obviously don't have to try and fix all of that right now here in this patch!

Reply:
I filed a bug:
https://github.com/mozilla/pdf.js/issues/15056